x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,784 @@
+
+   j. Repeat these steps for each additional node in your cluster.
+
+. After generating a certificate for each of your nodes, enter a password for
+   your private key when prompted.
+
+. Unzip the generated `elasticsearch-ssl-http.zip` file. This compressed file
+   contains one directory for both {es} and {kib}. Within the `/elasticsearch`
+   directory is a directory for each node that you specified with it's own
+   `http.p12` file. For example:
++
+--
+[source,txt]
+----
+/node1
+|_ README.txt
+|_ http.p12
+|_ sample-elasticsearch.yml
+----
+
+[source,txt]
+----
+/node2
+|_ README.txt
+|_ http.p12
+|_ sample-elasticsearch.yml
+----
+
+[source,txt]
+----
+/node3
+|_ README.txt
+|_ http.p12
+|_ sample-elasticsearch.yml
+----
+--
+
+. If necessary, rename the `http.p12` file to match the name of your existing
+certificate for HTTP client communications. This keystore contains both a
+certificate and key.
+
+. [[start-rolling-restart-http,step 5]]On the current node in your cluster where you're updating the keystore,
+start a <<restart-cluster-rolling,rolling restart>>.
++
+Stop at the step indicating *Perform any needed changes*, and then proceed to
+the next step in this procedure.
+
+. Replace your existing keystore with the new keystore, ensuring that the
+file names match. For example, `node1-http.p12`.
++
+If your 
+<<cert-password-updates,keystore password is changing>>, then save the
+keystore with a new filename so that {es} doesn't attempt to reload the file
+before you update the password.
+
+. If you needed to save the new keystore with a new filename, update the
+`ES_PATH_CONF/elasticsearch.yml` file to use the filename of the new keystore.
+For example:
++
+[source,yaml]
+----
+xpack.security.http.ssl.enabled: true
+xpack.security.http.ssl.keystore.path: node1-http.p12
+----
+
+. If your keystore password is changing, add the password for your private key
+to the secure settings in {es}.
++
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.http.ssl.keystore.secure_password
+----
+
+. Start the node where you updated the keystore, and then complete the remaining
+steps for a <<restart-cluster-rolling,rolling restart>>.
+
+. [[verify-keystore-http,step 10]](Optional) Use the <<security-api-ssl,SSL certificate API>> to verify that
+{es} loaded the new keystore.
+
+. One node at a time, complete <<start-rolling-restart-http>> through
+<<verify-keystore-http>> until you've updated all keystores in your cluster.
+
+[[update-node-certs-different]]
+=== Update security certificates with a different CA
+++++
+<titleabbrev>With a different CA</titleabbrev>
+++++
+If you have to trust a new CA, generate a new CA certificate and key for each
+node that is signed by the new CA certificate and instruct your nodes to trust
+this certificate chain.
+
+[[node-certs-different-transport]]
+==== Generate a new certificate for the transport layer
+Create a new CA certificate and add it to your existing CA truststore. After
```

Comment:
it's actually "After you have finished with updating your certificates for all nodes, you can remove the old CA certificate from your truststore" In that light, maybe not even mention it here and add it as a last step